feat(apple): Rename the OOM integration (#5938)
getsentry/sentry-docs: src/platforms/apple/common/configuration/out-of-memory.mdx

---
title: Out Of Memory
sidebar_order: 10
description: "Learn how to turn off Out Of Memory tracking"
---

> <Note>
> This integration is only available before 8.0.0. We renamed it to <PlatformLink to="/configuration/watchdog-terminations/">watchdog terminations</PlatformLink> in the 8.0.0 release.
> </Note>

Reviewer: Maybe this is obvious, but can we be more explicit about what we're talking about here? The SDK version? The iOS version?
```suggestion
<Note>

This integration is only available before version 8.0.0. It's now called <PlatformLink to="/configuration/watchdog-terminations/">Watchdog Terminations</PlatformLink> in the 8.0.0 release.

</Note>
```

Author: It's the SDK version, which I don't know how we call that in the docs. `sentry-cocoa`? `The Sentry SDK`?

Reviewer: I think just saying "version 8.0.0 of the Sentry Apple SDK" is fine.

When a typical unhandled error occurs, the Apple SDK writes a report to disk with the current state of your application, with details like the stack trace, tags, breadcrumbs, and so on, before the app terminates. With OOM crashes, the operating system can kill your app without further notice, which means the SDK can't write a report to disk.
As a result, in the Apple SDK, we track OOM crashes during the app start based on heuristics, but getting the state of the app when an OOM occurs is challenging. To provide the same state as we do for a typical crash, we would periodically need to store the app state to disk. This would cause a lot of I/O, which in turn could negatively affect the application's performance. To avoid this, OOM events have less context than typical unhandled errors.

When a user launches the app, the SDK checks the following statements and reports an OOM if all of them are true:

- The app didn't crash on the previous run.
- The app was in the foreground/active.
- The user launched the app at least once. When your app produces an OOM during the first launch, the SDK can't detect it.

> - There was no debugger attached to the process of the app. If there was, our logic would falsely report OOMs whenever you restarted the app in development.

Reviewer: ```suggestion
- There was no debugger attached to the process of the app. (If there was, our logic would falsely report OOMs whenever you restarted the app in development.)
```

Reviewer: Just not super clear why you committed the change for the other instance of this but not this one. It's pretty much the same language.

Author: Just missed it.
